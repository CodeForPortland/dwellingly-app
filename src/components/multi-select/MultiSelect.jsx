import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Search from '../search/Search';

class MultiSelect extends Component {
  constructor(props) {
    super(props)
    this.state = {
      selected: []
    }

    this.handleUpdatingSelected = this.handleUpdatingSelected.bind(this);
  }

  handleUpdatingSelected(data) {
    this.setState(prevState => ({selected: [...prevState.selected, data]}))
  }

  render() {
    const { placeholder, data, onSearchSelection } = this.props;
    return (
      <div>
        <Search
          onSearchSelection={onSearchSelection}
          onUpdatingSelected={this.handleUpdatingSelected}
          placeholder={placeholder}
          searchData={data}
          multiple />
      </div>
    )
  }
}

MultiSelect.propTypes = {
  placeholder: PropTypes.string,
  data: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  onSearchSelection: PropTypes.func.isRequired
}
MultiSelect.defaultProps = {
  placeholder: undefined
}

export default MultiSelect;
