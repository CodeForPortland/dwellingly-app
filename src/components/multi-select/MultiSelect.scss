@import '../../constants/constants.scss';
.selectedOptions {
  display: flex;
  margin-top: 20px;
  & .selectedPill {
    display: flex;
    align-items: center;
    padding: 3px 10px;
    border-radius: 18px;
    background: $color__gray--light;
    & .icon {
      font-size: .7em;
      margin-left: 8px;
      margin-top: 1px;
    }
  }
}
