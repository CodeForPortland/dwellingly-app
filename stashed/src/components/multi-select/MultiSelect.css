/* V A R I A B L E S */
/* end variables */
/* F U N C T I O N S */
/* end functions */
/* M I X I N S */
/* end mixins */
.selectedOptions {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  margin-top: 20px;
}

.selectedOptions .selectedPill {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  padding: 3px 10px;
  margin: 0 0.3rem;
  border-radius: 18px;
  background: #eeeeee;
}

.selectedOptions .selectedPill:first-child {
  margin-left: 0;
}

.selectedOptions .selectedPill .pillClose {
  cursor: pointer;
  font-size: .7em;
  margin-left: 8px;
  margin-top: 1px;
}
/*# sourceMappingURL=MultiSelect.css.map */