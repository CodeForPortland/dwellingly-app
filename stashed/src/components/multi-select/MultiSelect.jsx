import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Search from '../search/Search';
import Icon from '../icon/Icon';

import './MultiSelect.scss'

class MultiSelect extends Component {
  constructor(props) {
    super(props)
    this.state = {
      selected: []
    }

    this.handleUpdatingSelected = this.handleUpdatingSelected.bind(this);
    this.handleDeletingSelected = this.handleDeletingSelected.bind(this);
  }

  handleUpdatingSelected(data) {
    if(!this.state.selected.includes(data)) {
      this.setState(prevState => ({selected: [...prevState.selected, data]}))
    } else {
      this.setState(prevState => ({selected: prevState.selected.filter((item) => item !== data)}))
    }
  }

  handleDeletingSelected() {
    this.state.selected.map((item) => this.state.selected.splice(item, 1));
    this.setState(prevState => ({selected: [...prevState.selected]}));
  }

  render() {
    const { placeholder, data, onSearchSelection, filterSubset } = this.props;
    return (
      <div>
        <Search
          multiSelectData={this.state.selected}
          onSearchSelection={onSearchSelection}
          onUpdatingSelected={this.handleUpdatingSelected}
          onTogglingOption={this.handleDeletingSelected}
          filterSubset={filterSubset}
          placeholder={placeholder}
          searchData={data}
          multiple />
          <div className="selectedOptions">
            {this.state.selected.map(({firstName, lastName, id}) =>
              <div className="selectedPill" key={id}>
                <span>{firstName} {lastName}</span>
                <span className="pillClose" onClick={this.handleDeletingSelected} role="presentation">
                  <Icon icon="close" />
                </span>
              </div>
            )}
          </div>
      </div>
    )
  }
}

MultiSelect.propTypes = {
  filterSubset: PropTypes.arrayOf(PropTypes.string).isRequired,
  placeholder: PropTypes.string,
  data: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  onSearchSelection: PropTypes.func.isRequired
}
MultiSelect.defaultProps = {
  placeholder: undefined
}

export default MultiSelect;
