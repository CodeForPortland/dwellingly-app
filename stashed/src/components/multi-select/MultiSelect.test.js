import React from 'react';
import renderer from 'react-test-renderer';
import MultiSelect from './MultiSelect';

describe('MultiSelect Component', () => {
  it('renders correctly', () => {
    const props = {
      filterSubset: [''],
      placeholder: '',
      data: [{}],
    }
    const MultiSelectComponent = renderer.create(<MultiSelect {...props} /> ).toJSON();
    expect(MultiSelectComponent).toMatchSnapshot();
  });
});
