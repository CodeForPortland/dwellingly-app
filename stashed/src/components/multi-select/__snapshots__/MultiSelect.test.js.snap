// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiSelect Component renders correctly 1`] = `
<div>
  <div
    className="searchContainer"
  >
    <div
      className="searchContainerInner"
    >
      <div
        className="align--left"
      >
        <div>
          <label>
            <input
              className="searchBar"
              onChange={[Function]}
              onFocus={[Function]}
              placeholder=""
              type="text"
              value=""
            />
          </label>
        </div>
      </div>
      
      <div
        className="selectedOptions"
      />
    </div>
  </div>
  <div
    className="selectedOptions"
  />
</div>
`;
